network/src/tests/receiver_tests.rs

```diff
@@ -49,3 +74,81 @@ async fn receive() {
     let received = message.unwrap();
     assert_eq!(received, sent);
 }
+
+#[tokio::test]
+async fn receive_no_ack() {
+    // Make the network receiver.
+    let address = "127.0.0.1:4001".parse::<SocketAddr>().unwrap();
+    let (tx, mut rx) = channel(1);
+    Receiver::spawn(address, TestHandlerNoAck { deliver: tx });
+    sleep(Duration::from_millis(50)).await;
+
+    // Send a message.
+    let sent = "Hello, world!";
+    let bytes = Bytes::from(bincode::serialize(sent).unwrap());
+    let stream = TcpStream::connect(address).await.unwrap();
+    let mut transport = Framed::new(stream, LengthDelimitedCodec::new());
+    transport.send(bytes.clone()).await.unwrap();
+
+    // Ensure the message gets passed to the channel.
+    let message = rx.recv().await;
+    assert!(message.is_some());
+    let received = message.unwrap();
+    assert_eq!(received, sent);
+}
+
+#[tokio::test]
+async fn accounting_receive() {
+    // Make the central accounting.
+    let mut central_accounting = CentralAccounting::<TestHandler>::new(1);
+
+    // Make the first accounting receiver.
+    let address_1 = "127.0.0.1:4003".parse::<SocketAddr>().unwrap();
+    let (tx_1, mut rx_1) = channel(1);
+    let test_handler_1 = TestHandler { deliver: tx_1 };
+    let accounting_receiver_1 = AccountingReceiver::new(address_1);
+
+    // Make the second accounting receiver.
+    let address_2 = "127.0.0.1:4002".parse::<SocketAddr>().unwrap();
+    let (tx_2, mut rx_2) = channel(1);
+    let test_handler_2 = TestHandler { deliver: tx_2 };
+    let accounting_receiver_2 = AccountingReceiver::new(address_2);
+
+    // Register the accounting receivers with the central accounting.
+    let (receiver_id_1, tx_message_1) = central_accounting.register(test_handler_1).await;
+    let (receiver_id_2, tx_message_2) = central_accounting.register(test_handler_2).await;
+
+    // Start the central accounting.
+    central_accounting.spawn().await;
+
+    // Start the accounting receivers with the provided receiverIDs and tx_forwarding channels.
+    accounting_receiver_1.spawn(receiver_id_1, tx_message_1);
+    accounting_receiver_2.spawn(receiver_id_2, tx_message_2);
+    sleep(Duration::from_millis(50)).await;
```

Comment:
Ah, I think eventually this can make the test flaky when we end up running in different environments. I know that things so far have been ok, but those assumptions could change (for example, why `50ms` and not `100ms`?) . 

I believe the way to go would be to make the code bellow - which tries to connect to the tcp - to retry a few times (with some intermediate sleep) before failing permanently. 

In any case, leaving like this for now is fine, but we might need to re-think the approach we take in general in such cases